onnxruntime/core/framework/kernel_type_str_resolver.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
This file contains multiple resolvers. Would be good to name it so. In fact I was searching for the default full build resolver in a separate file, so I guess it would be better to keep the default (full build) resolver in a separate file and name it KernelTypeStrResolver and the one specific to ort format models as OrtFormatKernelTypeResolver. Having separate files probably makes it easier to exclude/include them in minimal builds at the cmake level?

Reply:
here is the way I think of it:

there is one main implementation of `IKernelTypeStrResolver`: `KernelTypeStrResolver`. It provides a resolve implementation and ways to populate (registration/deserialization) and save its state.

`AutoRegisteringKernelTypeStrResolver` is just a wrapper around a `KernelTypeStrResolver` which provides behavior similar to the original kernel matching - resolving using op schemas on demand.

I'll keep this suggestion in mind. perhaps we can also do things differently if we don't pass IKernelTypeStrResolver to the EPs and use a more opaque lookup functor instead.

Reply:
The name "AutoRegisteringKernelTypeStrResolver" is super confusing; it implies something is getting automatically registered. Please change the name to OnnxKernelTypeStrResolver or OpSchemaKernelTypeStrResolver or a better name to make it obvious.

Reply:
went with renaming to OpSchemaKernelTypeStrResolver.
it seems reasonable to me to have these classes in one header. it's one main class, KernelTypeStrResolver, and some related ones.